Marlin/src/module/endstops.cpp

```diff
@@ -608,10 +616,19 @@ void Endstops::update() {
   #endif
 
   // When closing the gap check the enabled probe
-  #if HAS_CUSTOM_PROBE_PIN
+  #if HAS_CUSTOM_PROBE_PIN && DISABLED(PROBE_ACTIVE_INPUT)
     UPDATE_ENDSTOP_BIT(Z, MIN_PROBE);
   #endif
 
+  #if ENABLED(PROBE_ACTIVE_INPUT)
+    #if HAS_CUSTOM_PROBE_PIN
+      UPDATE_ENDSTOP_BIT(Z, MIN);
+      SET_BIT_TO(live_state, _ENDSTOP(Z, MIN), ((READ(_ENDSTOP_PIN(Z, MIN)) != _ENDSTOP_INVERTING(Z, MIN_PROBE) && (READ(PROBE_ACTIVE_INPUT_PIN) == PROBE_ACTIVE_INPUT_STATE))));
+    #else
+      SET_BIT_TO(live_state, _ENDSTOP(Z, MIN), ((READ(_ENDSTOP_PIN(Z, MIN)) != _ENDSTOP_INVERTING(Z, MIN) && (READ(PROBE_ACTIVE_INPUT_PIN) == PROBE_ACTIVE_INPUT_STATE))));
+    #endif
+  #endif
```

Comment:
This is the only section of code I'm still unsure about.

I'm not sure why `UPDATE_ENDSTOP_BIT` is needed in the first case but not the other.

It's too late for me to try to make sense of it right now...

Reply:
If it is using a dedicated probe pin then use the standard function on the z minimum end stop, and new function on probe pin.

Reply:
@InsanityAutomation it looked like this might not have co-existed properly with multiple Z_MIN endstops. That sounds like a weird configuration, but since when has that stopped anybody?

I've reworked it in a way that I think should work, while also eliminating some duplication.

This has the caveat that you cannot use Z_MIN for a probe while also having Z_MIN endstops. I suspect that was already true anyway.

Please review 61a1fb7 to see if my changes look correct to you.